### PR TITLE
Bug fix in getDate function

### PR DESCRIPTION
**Pull request by eetiez**
Bug fix in getDate function : while loop condition has to include day=1.

### CODE REVIEW

**Comment by viiru-** [comment]
Please add a test as well. There are already a couple of tests for the Date class in pytrainer/test/lib/test_date.py that can be used as a base.

**Comment by eetiez** [comment]
I add a comment in pytrainer/lib/date.py as it was not clear for me (and I guess for others) why the function was returning "month + 1".

I also add a new test for getDate function to test every day in a year and make sure that the function returns a valid date.

File: pytrainer/test/lib/test_date.py [hunk 1]
```diff
@@ -65,7 +65,23 @@ def test_getDate_should_raise_ValueError_if_day_is_before_first(self):
...
+            
+    def test_getDate_should_return_valid_date_if_date_is_valid(self):
+        mock_calendar = Mock()
+        start_date = datetime.date(2019, 1, 1)
+        # list of every day in 2019
+        date_list = [start_date + datetime.timedelta(days=x) for x in range(0, 365)]
+        end_date = datetime.date(2019, 12, 31)
+        # test every day in the list
+        for date in date_list:
+           with self.subTest(date=date):
```
Comment (viiru-) [inline pytrainer/test/lib/test_date.py]: Sadly subTest isn't supported on Python 2.7 (which we still need to support at least for this release). Can you remove that, and rebase on top of current master (which fixes issues with new versions of the mock library)?

**Comment by heiderich** [comment]
I would suggest to also add a test for "invalid dates" if this makes sense, as this seems to be the main use of this function if I understand

https://github.com/pytrainer/pytrainer/blob/9ac119c08d6ca506b29402a033b345c7585a3614/pytrainer/lib/date.py#L91-L92

correctly.

Maybe it would also be worth to test several years including at least one leap year and one non-leap year.

**Comment by eetiez** [comment]
I was indeed using python 3... Sorry. I replaced subTest with a try/except. However, tests stop at the first fail whereas every tests were done with subTest. I left the line in comment.

I also added 2020 (leap year) in the list of dates for the test.

Obviously, I merged your last commit before my commit.

**Comment by viiru-** [comment]
Thanks a lot @eetiez, merged.

**Pull request merged**